packages/solid/src/use-chat.ts

```diff
@@ -54,15 +62,111 @@ export type UseChatHelpers = {
   input: Accessor<string>;
   /** Signal setter to update the input value */
   setInput: Setter<string>;
+  /** An input/textarea-ready onChange handler to control the value of the input */
+  handleInputChange: (
+    e:
+      | (Event & { currentTarget: HTMLInputElement })
+      | (Event & { currentTarget: HTMLTextAreaElement }),
+  ) => void;
   /** Form submission handler to automatically reset input and append a user message */
-  handleSubmit: (e: any, chatRequestOptions?: ChatRequestOptions) => void;
+  handleSubmit: (
+    e: Event & { currentTarget: HTMLFormElement },
+    chatRequestOptions?: ChatRequestOptions,
+  ) => void;
   /** Whether the API request is in progress */
   isLoading: Accessor<boolean>;
   /** Additional data added on the server via StreamData */
   data: Accessor<JSONValue[] | undefined>;
 };
 
-let uniqueId = 0;
+const getStreamedResponse = async (
+  api: string,
+  chatRequest: ChatRequest,
+  mutate: (data: Message[]) => void,
+  setStreamData: Setter<JSONValue[] | undefined>,
+  streamData: Accessor<JSONValue[] | undefined>,
+  extraMetadata: any,
+  messagesRef: Message[],
+  abortController: AbortController | null,
+  generateId: IdGenerator,
+  streamMode?: 'stream-data' | 'text',
+  onFinish?: (message: Message) => void,
+  onResponse?: (response: Response) => void | Promise<void>,
+  onToolCall?: UseChatOptions['onToolCall'],
+  sendExtraMessageFields?: boolean,
+) => {
+  // Do an optimistic update to the chat state to show the updated messages
+  // immediately.
+  const previousMessages = messagesRef;
+  mutate(chatRequest.messages);
+
+  const constructedMessagesPayload = sendExtraMessageFields
+    ? chatRequest.messages
+    : chatRequest.messages.map(
+        ({
+          role,
+          content,
+          name,
+          data,
+          annotations,
+          toolInvocations,
+          function_call,
+          tool_calls,
+          tool_call_id,
+        }) => ({
+          role,
+          content,
+          ...(name !== undefined && { name }),
+          ...(data !== undefined && { data }),
+          ...(annotations !== undefined && { annotations }),
+          ...(toolInvocations !== undefined && { toolInvocations }),
+          // outdated function/tool call handling (TODO deprecate):
+          tool_call_id,
+          ...(function_call !== undefined && { function_call }),
+          ...(tool_calls !== undefined && { tool_calls }),
```

Comment:
if these were not there before, let's not add them. the implementation is deprecated and broken